04-structs/models/trainers.go

```diff
@@ -0,0 +1,37 @@
+package models
+
+import (
+	"errors"
+	"fmt"
+)
+
+type Trainer struct {
+	Name  string
+	Party []*Pokemon
+}
+
+const MaxPartySize int = 6
+
+func (t *Trainer) AddToParty(p *Pokemon) error {
+	if len(t.Party) >= MaxPartySize {
+		return errors.New("o time do treinador já está cheio")
+	}

```

Comment:
Lo mejor seria validar si el parametro es nil, en este caso no lo esas validando y el append de un nil NO da un panic y si te agrega el item al slice, lo cual puede generar confusione o porlbmeas si luego accedes a este elemento asumiendo que no es nil.

```suggestion
func (t *Trainer) AddToParty(p *Pokemon) error {
	if p == nil {
		return errors.New("pokemon não pode ser nulo")
	}

	if len(t.Party) >= MaxPartySize {
		return errors.New("o time do treinador já está cheio")
	}

```